aioxmpp/service.py

```diff
@@ -263,14 +258,73 @@ def __get__(self, instance, owner):
         return obj
 
 
+class DependencyGraphNode:
+    def __init__(self):
+        self.class_ = None
+
+    def __repr__(self):
+        return "DependencyGraphNode({!r})".format(self.class_)
+
+
+class DependencyGraph:
+
+    def __init__(self, edges=None, nodes=None):
+        if edges is None:
+            edges = []
+        if nodes is None:
+            nodes = []
+        self._edges = set(edges)
+        self._nodes = set(nodes)
+
+    def copy(self):
+        return DependencyGraph(self._edges, self._nodes)
```

Comment:
I know that it’s not a full deepcopy. But not doing a full deepcopy is exactly those cases where you’d normally overwrite ``__deepcopy__`` (e.g. where you have references to other things, for example classes, you don’t want to duplicate).